plugins/blockvolume/blockprovider/gluster-virtblock/gluster_virtblock.go

```diff
@@ -97,60 +99,75 @@ func (g *GlusterVirtBlk) CreateBlockVolume(name string, size uint64, hostVolume
 		logger.WithError(err).Error("failed in updating hostvolume _block-hosting-available-size metadata")
 	}
 
+	volInfo, err := volume.GetVolume(hostVolume)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get host volume info %s: %+v", hostVolume, err)
+	}
+	key := volume.BlockPrefix + name
+	val := strconv.FormatUint(size, 10)
+	volInfo.Metadata[key] = val
+	if err := volume.AddOrUpdateVolume(volInfo); err != nil {
+		log.WithError(err).Error("failed in updating volume info to store")
+	}
+
 	return &BlockVolume{
 		hostVolume: hostVolume,
 		name:       name,
 		size:       size,
 	}, err
 }
 
+func delBlockEntry(hostName string, name string) error {
+	hostVol, err := volume.GetVolume(hostName)
+	if err != nil {
+		return errors.New("block volume not found")
+	}
+
+	for k, _ := range hostVol.Metadata {
+		if k == (volume.BlockPrefix + name) {
+			delete(hostVol.Metadata, k)
+			if err := volume.AddOrUpdateVolume(hostVol); err != nil {
+				log.WithError(err).Error("failed in updating volume info to store")
+			}
+		}
+	}
+	return nil
+}
+
 // DeleteBlockVolume deletes a gluster block volume of give name
 func (g *GlusterVirtBlk) DeleteBlockVolume(name string, options ...blockprovider.BlockVolOption) error {
 	var (
 		blockVolOpts = &blockprovider.BlockVolumeOptions{}
-		hostVol      string
 	)
 
 	blockVolOpts.ApplyOpts(options...)
 
-	// TODO: Listing all the block volumes to delete one block vol will bottleneck at scale. Possible options:
-	// - Let block delete carry the host volume(optionally). The caller needs to keep this info returned in create vol, and send it in delete req.
-	// - Build a map in memory ([blockvolume]hostvolume)during init(or lazy) during init of provider/create of block volume
-	blockVols := g.BlockVolumes()
-
-	for _, blockVol := range blockVols {
-		if blockVol.Name() == name {
-			hostVol = blockVol.HostVolume()
-			break
-		}
-	}
-
-	if hostVol == "" {
+	blkVol, err := g.GetBlockVolume(name)
+	if err != nil || blkVol == nil {
 		return errors.New("block volume not found")
```

Comment:
define as constant it will be helpful in CSI to return success in case block volume not found

Reply:
define as constant, didn't understand?

Reply:
https://github.com/gluster/glusterd2/blob/master/pkg/errors/error.go